pandas/_libs/index.pyx

```diff
@@ -415,12 +406,12 @@ cdef class DatetimeEngine(Int64Engine):
             if not self.is_unique:
                 return self._get_loc_duplicates(val)
             values = self._get_index_values()
-            conv = _to_i8(val)
```

Comment:
isn't this a cimport?

Reply:
At the moment it is `cdef`d in `_libs.index`.  This replaces it with a call to `tslib.pydt_to_i8`.  As noted in the OP, `pydt_to_i8` is currently just `def`, but in the PR is made `cpdef` so that the c version can be called here (`tslib` is cimported into `_libs.index`)